tests/merchOrder.test.ts

```diff
@@ -84,6 +84,65 @@ describe('merch orders', () => {
       .called();
   });
 
+  test('members can only access store with a valid acm or ucsd email', async () => {
+    const conn = await DatabaseConnection.get();
+    const UCSDMember = UserFactory.fake({ credits: 10000 });
+    const ACMMember = UserFactory.fake({
+      email: 'gZhao@acmucsd.org',
+      credits: 10000,
+    });
+    const invalidMember = UserFactory.fake({
+      email: 'gZhao@gmail.com',
+      credits: 10000,
+    });
+
+    const affordableOption1 = MerchFactory.fakeOption({
+      quantity: 5,
+      price: 2000,
+      discountPercentage: 0,
+    });
+    const pickupEvent = MerchFactory.fakeFutureOrderPickupEvent();
+
+    await new PortalState()
+      .createUsers(ACMMember, UCSDMember, invalidMember)
+      .createMerchItemOptions(affordableOption1)
+      .createOrderPickupEvents(pickupEvent)
+      .write();
+
+    const emailService = mock(EmailService);
+    when(emailService.sendOrderConfirmation(ACMMember.email, ACMMember.firstName, anything()))
+      .thenResolve();
+    when(emailService.sendOrderConfirmation(UCSDMember.email, UCSDMember.firstName, anything()))
+      .thenResolve();
+
```

Comment:
For merch store access, we wouldn't even want non-UCSD students to be able to look at the items, let alone order them. So instead of checking that they can't place orders, can we check that they can't view collections? E.g. the `GET /collections` call throws `ForbiddenError`. That mimics what we're actually testing a bit better, plus it makes the test a lot more simpler